pandas/core/arrays/arrow/array.py

```diff
@@ -1478,7 +1478,13 @@ def to_numpy(
         return result
 
     def map(self, mapper, na_action: Literal["ignore"] | None = None):
-        if is_numeric_dtype(self.dtype):
+        if not is_string_dtype(self.dtype) and any(
+            [
+                is_numeric_dtype(self.dtype)
+                or pa.types.is_date(self.dtype.pyarrow_dtype)
+                or pa.types.is_timestamp(self.dtype.pyarrow_dtype)
+            ]
+        ):
```

Comment:
any idea why the super().map method isn't working correctly?  If fixing things there is viable, that would be a better outcome